src/pip/_internal/commands/install.py

```diff
@@ -226,6 +228,30 @@ def add_options(self):
     @with_cleanup
     def run(self, options, args):
         # type: (Values, List[str]) -> int
+        # Warn if running pip with sudo
+        try:
+            if sys.platform == 'darwin' or sys.platform == 'linux':
+                is_sudo = os.getuid() == 0
+            elif sys.platform == 'win32' or sys.platform == 'cygwin':
+                is_sudo = ctypes.windll.shell32.IsUserAnAdmin() != 0
+        except AttributeError:
+            is_sudo = False
+
+        is_venv = (hasattr(sys, 'real_prefix')
+                   or (hasattr(sys, 'base_prefix')
+                       and sys.base_prefix != sys.prefix))
+
+        if is_sudo and not is_venv and not options.root_path:
+            command = os.path.basename(sys.argv[0])
+            if command == "__main__.py":
+                command = os.path.basename(sys.executable) + " -m pip"
+            sudo_warning = ("Running pip as root is not recommended! "
+                            "This could result in the corruption of the "
+                            "package directories when a package manager is "
+                            "used. Try `%s install --user` instead."
+                            % command)
```

Comment:
It’s probably a good idea to use different messages on Windows and POSIX, since “root” does not mean anything on Windows (and the check was not looking for root in the first place).

The final “Try ...” message is problematic as well since it is almost impossible to display a perfect message here, see https://github.com/pypa/pip/pull/7532#discussion_r362325906. We should try our best avoid showing a comment if possible. I would simply say something like “Consider using --user instead”.

Finally, the same check should be applied to `uninstall` as well IMO (without a recommendation, just warn the user this may break the system unless you are sure what you’re doing).

Reply:
> Consider using --user instead

This is unnecessary, given the fix for #1668 automatically falls back to the user folders if the global location is unwritable.

Reply:
Sounds good - we can substitute "root" for "Administrator" for Windows, and take out `Try %s install --user`

I'll update this PR on Monday

Reply:
Actually, on Windows, running pip as administrator is fine, as there is no "system managed" Python package anyway - so installing via pip as administrator is the correct way of updating your system environment.

So I'd suggest restricting this warning to Unix systems only.

Reply:
Made some changes:
- Added `Consider using --user instead.` from @uranusjr 's suggestion
- Added this sudo check to uninstall from @uranusjr 's suggestion
- Disabled this check from Windows from @pfmoore 's suggestion